model/src/main/kotlin/LicenseFinding.kt

```diff
@@ -51,9 +52,41 @@ data class LicenseFinding(
 ) : Comparable<LicenseFinding> {
     companion object {
         private val COMPARATOR = compareBy<LicenseFinding>({ it.license.toString() }, { it.location })
+
+        /**
+         * Create a [LicenseFinding] with [detectedLicenseMapping]s applied.
+         */
+        fun createAndMap(
+            license: String,
+            location: TextLocation,
+            score: Float? = null,
+            detectedLicenseMapping: Map<String, String>
+        ): LicenseFinding = LicenseFinding(
+            license = if (license.isSpdxExpression() || detectedLicenseMapping.isEmpty()) {
+                license
+            } else {
+                license.applyDetectedLicenseMapping(detectedLicenseMapping)
+            }.toSpdx(),
+            location = location,
+            score = score
+        )
     }
 
     constructor(license: String, location: TextLocation, score: Float? = null) : this(license.toSpdx(), location, score)
 
     override fun compareTo(other: LicenseFinding) = COMPARATOR.compare(this, other)
 }
+
+/**
+ * Apply [detectedLicenseMapping] from the [org.ossreviewtoolkit.model.config.ScannerConfiguration] to any license
+ * String.
+ */
+private fun String.applyDetectedLicenseMapping(detectedLicenseMapping: Map<String, String>): String {
+    var result = this
+    detectedLicenseMapping.forEach { (from, to) ->
+        // Replace a license restricted by whitespace boundaries.
+        result = result.replace("""(?<!\S)${Regex.escape(from)}""".toRegex(), to)
+    }
+
+    return result
```

Comment:
Nit: Looks like this could be simplified via `apply {}`.

Reply:
Not sure how to simplify this with `apply {}`. Can you give me a hint what you have in mind?

Reply:
It doesn't work as I thought it would, please disregard.

Reply:
But I have another idea 😁 Maybe use `fold` similar to

https://github.com/oss-review-toolkit/ort/blob/787b903c760deba18ae311c3c17a3e132d1166f9/scanner/src/main/kotlin/scanners/scancode/ScanCodeResultParser.kt#L264-L271

Reply:
IMO, this is less readable. Is it ok, if I keep it as-is?

Reply:
`fold` is not available for a `Map`.

Reply:
Is it ok if I keep it as-is?

Reply:
Sure, I'm just commenting from the off 😉 Your code just remembered me of that code, which mostly does the same thing.